src/tests/CxAuthCall.test.ts

```diff
@@ -1,44 +1,110 @@
+params.set(CxParamType.FILTER, "*.ts");
+const auth = new CxAuth(cxScanConfig);
 
 describe("ScanCreate cases",() => {
-it('ScanCreate Successful case', async () => {
+    it('ScanCreate Successful case wait mode', async () => {
     const data = await auth.scanCreate(params);
-    const cxScanObject: CxScan = JSON.parse(data);
-    expect(cxScanObject.Status).toContain("Queued");
+    const cxCommandOutput: CxCommandOutput =JSON.parse(JSON.stringify(data))
+        if(cxCommandOutput.scanObjectList.length !== 0) {
+            const scanID = cxCommandOutput.scanObjectList.pop().ID
+            const scanObject = await auth.scanShow(scanID);
+            if(scanObject !== null && scanObject !== undefined) {
+                const status = scanObject.scanObjectList.pop().Status
+                expect(status).toMatch(/(Complete|Running|Queued)/i)
+            }
+        }
+        else {
+            expect(cxCommandOutput.exitCode).toEqual(0)
+        }
 })
+    it('ScanCreate Successful case no wait mode', async () => {
+        params.set(CxParamType.ADDITIONAL_PARAMETERS, "--nowait");
+        const data = await auth.scanCreate(params);
+        const cxCommandOutput: CxCommandOutput =JSON.parse(JSON.stringify(data))
+        if(cxCommandOutput.scanObjectList.length !== 0) {
+            const scanID = cxCommandOutput.scanObjectList.pop().ID
+            const scanObject = await auth.scanShow(scanID);
+            if(scanObject !== null && scanObject !== undefined) {
+                const status = scanObject.scanObjectList.pop().Status
+                expect(status).toMatch(/(Complete|Running|Queued)/i)
+            }
+        }
+        else {
+            expect(cxCommandOutput.exitCode).toEqual(0)
+        }
+    })
+
+    it('ScanCreate Successful case with Branch', async () => {
+        params.set(CxParamType.BRANCH, "main");
+        const data = await auth.scanCreate(params);
+        const cxCommandOutput: CxCommandOutput =JSON.parse(JSON.stringify(data))
+        if(cxCommandOutput.scanObjectList.length !== 0) {
+            const scanID = cxCommandOutput.scanObjectList.pop().ID
+            const scanObject = await auth.scanShow(scanID);
+            if(scanObject !== null && scanObject !== undefined) {
+                const status = scanObject.scanObjectList.pop().Status
+                //expect(status).toContain("Completed" || "Running" || "Queued")
+                expect(status).toMatch(/(Complete|Running|Queued)/i)
+            }
+        }
+        else {
+            expect(cxCommandOutput.exitCode).toEqual(0)
+        }
+
+    })
+
+});
+
+describe("ScanCreate cases",() => {
+    it('ScanCreate Failure case', async () => {
+        params.set(CxParamType.SAST_PRESET_NAME, "Checkmarx Default Jay");
+        const data = await auth.scanCreate(params);
+        const cxCommandOutput: CxCommandOutput =JSON.parse(JSON.stringify(data))
+        if(cxCommandOutput.scanObjectList.length !== 0) {
```

Comment:
We should just check if the exitcode is 1, it should always be one in this scenario